libbeat/processors/registry.go

```diff
@@ -0,0 +1,28 @@
+package processors
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type ProcessorInterface interface {
+	Process(event common.MapStr) (common.MapStr, error)
```

Comment:
I'm tempted to rename `Process` to `Exec` or `Run`, which sounds more generic to me. Plus we can write `Processor.Exec` instead of `Processor.Process`.